feat(send-signal): new documentation page relative to the feature send signal (#1890)
Scalingo/documentation: src/_posts/platform/app/2000-01-01-send-signal.md

$ scalingo --app my-app ps
+-------+---------+-------------+------+---------------------+
| NAME  | STATUS  |   COMMAND   | SIZE |     CREATED AT      |
+-------+---------+-------------+------+---------------------+
| web-1 | running | sample-test | M    | 2023/01/05 09:24:33 |
| web-2 | running | sample-test | M    | 2023/01/05 09:24:33 |
+-------+---------+-------------+------+---------------------+

$ scalingo --app my-app send-signal --signal SIGUSR1 web-1
-----> Sent signal 'SIGUSR1' to 'web-1' container.
```
An API endpoint is available to send a signal. More information about the endpoint [here](https://developers.scalingo.com/apps#send-signal-to-a-container).

## Which process catches the signal?

When transmitted to the container, the signal is forwarded to the first process that started your application, also called the entry point. This entry point can be defined in the [Procfile](https://doc.scalingo.com/platform/getting-started/heroku-compatibility#procfile).  

### Example of a Node.js Express application

The default command run by Scalingo for a Node.js application is `npm start`. It runs a command specified in the `.scripts.start` field of the `package.json`.  

> When a signal is sent to the application, `npm start` is the first process to catch it. It does not forward the signal to the child process started with the `.scripts.start` field.  

Reviewer: suggestion: I'm not fond of the word "It" as it may be confusing about what you are referring to. I suggest this rephrase but you are free to use it or not.

```suggestion
When a signal is sent to the application, `npm` is the first process to catch it. `npm` does not forward the signal to the child process started with the `.scripts.start` field.  
```

---

> title: Send a signal to a container

Reviewer: ```suggestion
title: Send a Signal to a Container
```

nav: Send Signal
modified_at: 2023-01-12 00:00:00
tags: app signal
---

## Introduction

Signals are standardized messages sent to a running program to trigger specific behavior.

On the Scalingo platform, sending signals is a useful way to trigger events while being the owner or collaborator of an application. This can be particularly useful for debugging purposes.

{% note %}

>   The platform handles only `SIGUSR1` and `SIGUSR2`.

Reviewer: I think that in English we prefer writing it that way:

```suggestion
  The platform only handles `SIGUSR1` and `SIGUSR2`.
```

{% endnote %}

## General purpose of the feature

> As said before, implementing a signal handler can be an easy and powerful way to debug applications.

Reviewer: Either you already said it and you don't need to say it again, or you think it's really important to have this sentence at the beginning of this section and you don't need to mention that you already said it.

```suggestion
```

At Scalingo we often use signals handler to:

- Dump heap allocations
- Dump all stack traces
- Print the number of Goroutines
- Print the memory consumption

> - etc…

Reviewer: Either you use `etc` or you use `...`. Not both.
```suggestion
- …
```

## How to send a signal to an application ?

To send a signal to an application, multiple possibilities are available:

- The CLI provides a `send-signal` command.

```sh

> $ scalingo -a my-app ps

Reviewer: We only use long options in the documentation:

```suggestion
$ scalingo --app my-app ps
```

+-------+---------+-------------+------+---------------------+
| NAME  | STATUS  |   COMMAND   | SIZE |     CREATED AT      |
+-------+---------+-------------+------+---------------------+
| web-1 | running | sample-test | M    | 2023/01/05 09:24:33 |
| web-2 | running | sample-test | M    | 2023/01/05 09:24:33 |
+-------+---------+-------------+------+---------------------+

$ scalingo -a my-app send-signal --signal SIGUSR1 web-1
-----> Sent signal 'SIGUSR1' to 'web-1' container.
```

- An API endpoint is available to send a signal. More information about the endpoint [here](https://developers.scalingo.com/apps#send-signal-to-a-container).

> ## Who catches the signal ?

Reviewer: I think that in English, "who" refers to a person. But in this case you are not talking about a person.

{% warning %}

>   When transmitted to the container, the signal is transmitted to the first process that started your application, also called the entry point.  

Reviewer: suggestion: There is a repetition of the word "transmitted".

```suggestion
  When transmitted to the container, the signal is forwarded to the first process that started your application, also called the entry point.  
```

>   Commonly, this entry point can be defined in the [Procfile](https://doc.scalingo.com/platform/getting-started/heroku-compatibility#procfile), so be careful on which process will catch the signal first.

Reviewer: - I'm not sure I understand what the "Commonly" word adds to the sentence. I would remove it.
- There is two sentences here, let's add a dot to breath a  bit while reading.
- I don't really like the wording "so be careful". It feels like you are threatening our customers :D 
- It is unclear what you want to warn people about by writing the second sentence

```suggestion
  Commonly, this entry point can be defined in the [Procfile](https://doc.scalingo.com/platform/getting-started/heroku-compatibility#procfile). This is important to have this fact in mind as the entry point may silently ignore the sent signal.
```

> {% warning %}
>   When transmitted to the container, the signal is transmitted to the first process that started your application, also called the entry point.  
>   Commonly, this entry point can be defined in the [Procfile](https://doc.scalingo.com/platform/getting-started/heroku-compatibility#procfile), so be careful on which process will catch the signal first.
> {% endwarning %}

Reviewer: I'm not sure this is a warning block. I think it's the content of the section. You describe which process receives the signal. 
```suggestion

When transmitted to the container, the signal is transmitted to the first process that started your application, also called the entry point.  
Commonly, this entry point can be defined in the [Procfile](https://doc.scalingo.com/platform/getting-started/heroku-compatibility#procfile), so be careful on which process will catch the signal first.

```

> For example, on a Node.js application, the default command run by Scalingo is `npm start`, it runs a predefined command specified in the "start" property of a package's "scripts" object (usually named package.json).  

Reviewer: - Suggestion to ease the reading. You have multiple sentences, split them with a dot.
- This is how we talk about the `start` property in the doc (https://doc.scalingo.com/languages/nodejs/start#nodejs-app-startup)

```suggestion
For example, the default command run by Scalingo for a Node.js application is `npm start`. It runs a command specified in the `.scripts.start` field of the `package.json`.
```
